apps/api/src/services/competition-manager.service.ts

```diff
@@ -558,87 +586,164 @@ export class CompetitionManager {
   }
 
   /**
-   * Get the leaderboard for a competition
+   * Retrieves the saved leaderboard from the database for an ended competition
    * @param competitionId The competition ID
-   * @returns Array of agent IDs sorted by portfolio value
+   * @returns The saved leaderboard entries or empty array if not found
    */
-  async getLeaderboard(competitionId: string) {
-    try {
-      // TODO: this function has many paths it potentially takes depending on
-      //  the state of several tables. We should refactor and ideally only have
-      //  one, which doesn't have to do calculations for every (uncached) request.
-      const competition = await findById(competitionId);
-      const competitionStatus = competition?.status;
+  private async getSavedLeaderboard(
+    competitionId: string,
+  ): Promise<LeaderboardEntry[]> {
+    const leaderboardEntries =
+      await findLeaderboardByTradingComp(competitionId);
+    return leaderboardEntries.map((entry) => ({
```

Comment:
This is more of a question than suggestion, but any reason we to this `map` to remove the extra entry fields vs. reducing the fields in the SQL select statement?